### DISCUSSION TITLE
Support for Scroll Area in Table Responsiveness.

### DISCUSSION THREAD

**Discussion by ruslanguns** [KevinVandy/mantine-react-table#3] [Ideas]
I would like to propose the addition of a scroll area feature for the table responsiveness of the table using the ScrollArea component from Mantine Core. This would allow for control of the area with custom scrollbars and would improve the user experience by allowing for easy navigation and control of large data sets.

I believe this feature would be of great interest and value to the community.

https://mantine.dev/core/scroll-area/

**Comment by KevinVandy** [comment]
Yeah sure, I had originally made the table body and table components just have overflows, but this is probably more stock Mantine. I'll look into it. Though it might not work with my current virtualization logic.

**Reply by ruslanguns** [reply]
Understood

**Reply by Toshinaki** [reply]
@KevinVandy 
Hi, any updates on this?
I use `ScrollArea` in all my scrollable components and pages, only table with default scrollbar looks inconsistent.

**Reply by KevinVandy** [reply]
Just use CSS to change the look of the scrollbars

**Reply by Toshinaki** [reply]
@KevinVandy 
I tried to copy the source code and changed the `<Box>` inside the container component to `<ScrollArea>`,
didn't test a lot but it just worked...
If it's about virtualization, if `enableColumnVitrualization` or `enableRowVitrualization` is `true`, use `<Box>`, else use `<ScrollArea>`. Would this work?

**Reply by sbasher314** [reply]
I have a POC fork adding support for the Mantine ScrollArea which also supports virtualization (into / against v2 as that is the version my company is using) -- Is there any interest in this feature officially @KevinVandy, if so I can clean it up and open a PR. This will obviously be a breaking change, so we could conditionally enable / disable the scrollArea (I was thinking a prop named `withScrollArea` like the Mantine `Select` components, and defaulting to `false` for back-compatibility so that we don't break pre-existing implementations)

Thoughts?

**Reply by KevinVandy** [reply]
Sure

**Reply by sbasher314** [reply]
Awesome, I've got a PR in for this feature here! https://github.com/KevinVandy/mantine-react-table/pull/485 Let me know if there's anything that I missed or if anything broke that I didn't catch
